Makefile

```diff
@@ -12,7 +12,46 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+.PHONY: push-multiarch-% push-multiarch
+
 CMDS=csi-node-driver-registrar
 all: build
 
 include release-tools/build.make
+
+# This target builds multiarch images using Moby BuildKit builder toolkit.
+# Docker Buildx is included in Docker 19.03 and needs DOCKER_CLI_EXPERIMENTAL enabled to run corresponding commands.
+# Currently amd, s390x and Windows manifest is pushed for canary, release branch and released tags.
+# Images generated from Prow build are pushed to staging area on gcr
+push-multiarch-%:
+	make BUILD_PLATFORMS="windows amd64 .exe"
+	set -ex; \
+	gcloud auth configure-docker; \
+	DOCKER_CLI_EXPERIMENTAL=enabled; \
+	export DOCKER_CLI_EXPERIMENTAL; \
+	docker buildx create --use --name multiarchimage-buildertest; \
+	pushMultiArch () { \
+                tag=$$1; \
+                docker buildx build --push -t $(IMAGE_NAME):amd64-linux-$$tag --platform=linux/amd64 -f $(shell if [ -e ./cmd/$*/Dockerfile.multiarch ]; then echo ./cmd/$*/Dockerfile.multiarch; else echo Dockerfile.multiarch; fi) --label revision=$(REV) .; \
+                docker buildx build --push -t $(IMAGE_NAME):s390x-linux-$$tag --platform=linux/s390x -f $(shell if [ -e ./cmd/$*/Dockerfile.multiarch ]; then echo ./cmd/$*/Dockerfile.multiarch; else echo Dockerfile.multiarch; fi) --label revision=$(REV) .; \
+                docker buildx build --push -t $(IMAGE_NAME):amd64-windows-$$tag --platform=windows -f $(shell if [ -e ./cmd/$*/Dockerfile.Windows ]; then echo ./cmd/$*/Dockerfile.Windows; else echo Dockerfile.Windows; fi) --label revision=$(REV) .; \
```

Comment:
Have you tried doing this with `Dockerfile.multiarch` instead of `Dockerfile.Windows`?

You must extract the target OS from TARGETPLATFORM instead of hard-coding `linux` and the .exe suffix may need special handling.

Reply:
>Have you tried doing this with Dockerfile.multiarch instead of Dockerfile.Windows?

I haven't tried this. We can't test windows images at our end. May be @jingxu97 can try this out.

Ideally instead of writing different `docker buildx build` command for each arch build and then creating manifest for them, single command with `--push --platform=linux/amd64,linux/s390x` should work. However I feel  we can try this out later, once current changes are working as expected.

Reply:
Since Dockerfile.multiarch contains a 'RUN' command, we cannot use it to build windows image on a linux machine with buildx.

Reply:
So we can build Windows binaries on the Linux host, but not inside a Docker container? Or does building for Windows need a Windows host?